### ISSUE TITLE
Remove connection timeout

### ISSUE THREAD

**Issue by AmarOk1412** [Corvusoft/restbed#265]
Hi,
by default connection timeout is set to `5 secs` there is a way to change the duration with `set_connection_timeout`, but there is no way to remove this timeout.

By the way, this doesn't work:

```cpp
std::chrono::seconds sec(std::chrono::seconds::max());
settings->set_connection_timeout(sec); // Will give -1000

// But this is ok
std::chrono::milliseconds ms(std::chrono::milliseconds::max());
settings->set_connection_timeout(ms); // high value
```

It's not a real issue because `std::chrono::milliseconds::max()` is very huge

**Comment by ben-crowhurst** [comment]
Hmmm, I'm surprised settings-set_connection_timeout(0) doesn't remove the timeout.  Will look to get a fix into 4.7.

**Comment by ben-crowhurst** [comment]
Setting the value to zero doesn't appear to have the desired effect on the underlying asio::steady_timer.

It would seem the only workaround is to set a large timeout interval that meets your current circumstance. :|
